Add-ons/UmbracoForms/Developer/Extending/Adding-a-Type-v7.md

```diff
@@ -0,0 +1,137 @@
+
+## Preparations
+
+Create a new ASP.NET or class project in Visual Studio add references to the Umbraco.Forms.Core.dll.
+
+## Adding the type to Forms
+
+The Forms API contains a collection of classes that the provider model automatically registers. So to add a new type to Forms you inherit from the right class. In the sample below we use the class for the workflow type.
+
+```csharp
+public class Class1 : Umbraco.Forms.Core.WorkflowType
+{
+    public override WorkflowExecutionStatus Execute(Umbraco.Forms.Core.Record record)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override List<Exception> ValidateSettings() {
+        throw new NotImplementedException();
+    }
+}
+```
+
+When you implement this class you get two methods added. One of them is Execute which performs the execution of the workflow and the other is a method which validates the workflow settings, we will get back to these settings later on.
+
+Even though we have the class inheritance in place, we still need to add a bit of default information.
+
+## Setting up basic type information
+
+Even though we have the class inheritance in place, we still need to add a bit of default information. This information is added in the class's empty constructor like this:
+
+```csharp
+public Class1() {
+    this.Name = "The logging workflow";
+    this.Id = new Guid("D6A2C406-CF89-11DE-B075-55B055D89593");
+    this.Description = "This will save an entry to the log";
+}
+```
+
+All three are mandatory and the ID must be unique, otherwise the type might conflict with an existing one.
+
+## Adding settings to a type
+
+Now that we have a basic class setup, we would like to pass setting items to the type. So we can reuse the type on multiple items but with different settings. To add a setting to a type, we add a property to the class, and give it a specific attribute like this:
+
+```csharp
+[Umbraco.Forms.Core.Attributes.Setting("Log Header",
+        description = "Log item header",
+        view = "TextField")]
+public string LogHeader { get; set; }
+```
+
+The Umbraco.Forms.Core.Attributes.Setting registers the property in Umbraco Forms and there will automatically be UI and storage generated for it. In the attribute, a name, description and the view to be rendered is defined.
+
+With the attribute in place, the property value is set every time the class is instantiated by Umbraco Forms. This means you can use the property in your code like this:
+
+```csharp
+[Umbraco.Forms.Core.Attributes.Setting("Document ID",
+        description = "Node the log entry belongs to",
+        view = "Pickers.Content")]
+public string Document { get; set; }
+
+public override Enums.WorkflowExecutionStatus Execute(Record record) {
```

Comment:
Tiny detail as your being specific with class names in your examples, make sure the one on line 20 is the same.

Reply:
Thanks - made both the full version (especially with the differnce between v7 & v8 and user can decide to leave as is or change to a using statement